java-frontend/src/test/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheckTest.java

```diff
@@ -78,4 +78,9 @@ public void checkEndOfExecution(CheckerContext context) {
       assertThat(atof.alwaysTrue()).isEmpty();
     }
   }
+
+  @Test
+  public void test_constraint_copy_SONARJAVA2351() throws Exception {
```

Comment:
While getting the reference to a ticket for further explanation is a nice touch, I would keep it as a comment and try to have a descriptive name for the test and its testfile : this can get obsolete at some point of time, indirection is painful to read, and it is rather cryptic here.